File: src/Continuous/Prog10/Vibration/Vibration.java
package Continuous.Prog10.Vibration;

import java.applet.Applet;
import java.awt.BorderLayout;
import java.awt.Label;
import java.util.Date;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import javax.swing.Timer;

@SuppressWarnings("serial")
public class Vibration extends Applet implements ActionListener {
   static final String kRunString = "Run";
   CFramePanel mFramePanel;
   CAnimationPanel mAnimPanel;
   Timer mTimer = null;
   long mLastTime = 0;

   public void init() {
      this.setLayout(new BorderLayout());
      this.mAnimPanel = new CAnimationPanel();
      this.add("Center", this.mAnimPanel);
      Label copyright = new Label("You need permission to use this program. © 2005 B J Stone, bjs@mech.uwa.edu.au.");
      this.add("South", copyright);
      this.validate();
   }

   public void start() {
      this.mTimer = new Timer(20, this);
      this.mTimer.start();
   }

   public void actionPerformed(ActionEvent e) {
      if (this.mTimer != null) {
         Date ddd = new Date();
         long thisTime = ddd.getTime();

         if (this.mLastTime == 0) {
            this.mLastTime = thisTime;
         }
         double realSeconds = (thisTime - this.mLastTime) / 1000.0;
         this.mLastTime = thisTime;

         this.mAnimPanel.SetRealParameter(0, realSeconds);
      }
   }

}

File: site/swingjs/j2s/Continuous/Prog10/Vibration/Vibration.js
(function(){var P$=Clazz.newPackage("Continuous.Prog10.Vibration"),I$=[[0,'java.awt.BorderLayout','Continuous.Prog10.Vibration.CAnimationPanel','java.awt.Label','javax.swing.Timer']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "Vibration", null, 'java.applet.Applet', 'java.awt.event.ActionListener');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
this.mTimer=null;
this.mLastTime=0;
},1);

C$.$fields$=[['J',['mLastTime'],'O',['mFramePanel','Continuous.Prog10.Vibration.CFramePanel','mAnimPanel','Continuous.Prog10.Vibration.CAnimationPanel','mTimer','javax.swing.Timer']]]

Clazz.newMeth(C$, ['init$','init'],  function () {
this.setLayout$java_awt_LayoutManager(Clazz.new_($I$(1,1)));
this.mAnimPanel=Clazz.new_($I$(2,1));
this.add$S$java_awt_Component("Center", this.mAnimPanel);
var copyright=Clazz.new_($I$(3,1).c$$S,["You need permission to use this program. \u00a9 2005 B J Stone, bjs@mech.uwa.edu.au."]);
this.add$S$java_awt_Component("South", copyright);
this.validate$();
});

Clazz.newMeth(C$, ['start$','start'],  function () {
this.mTimer=Clazz.new_($I$(4,1).c$$I$java_awt_event_ActionListener,[20, this]);
this.mTimer.start$();
});

Clazz.newMeth(C$, ['actionPerformed$java_awt_event_ActionEvent','actionPerformed'],  function (e) {
if (this.mTimer != null ) {
var ddd=Clazz.new_(java.util.Date);
var thisTime=ddd.getTime$();
if (Long.$eq(this.mLastTime,0 )) {
this.mLastTime=thisTime;
}var realSeconds=(Long.$sub(thisTime,this.mLastTime)) / 1000.0;
this.mLastTime=thisTime;
this.mAnimPanel.SetRealParameter$I$D(0, realSeconds);
}});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('5.0.1-v7');//Created 2025-10-14 19:43:49 Java2ScriptVisitor version 5.0.1-v7 net.sf.j2s.core.jar version 5.0.1-v7
